lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -238,27 +238,72 @@ void SessionState::set_current_version(
   if (session_uc) {
     session_uc->is_current_version_updated = true;
     session_uc->updated_current_version    = new_session_version;
+    MLOG(MDEBUG) << " Current version is " << get_current_version();
   }
-  MLOG(MINFO) << " Current version is " << get_current_version();
 }
+
 /* Add PDR rule to this rules session list */
-void SessionState::insert_pdr(SetGroupPDR* rule) {
-  PdrList_.push_back(*rule);
+void SessionState::insert_pdr(
+    SetGroupPDR* rule, bool crit_add, SessionStateUpdateCriteria* session_uc) {
+  // Check if it already exists
+  int32_t Pdr_index;
+  Pdr_index = get_pdr_index(rule->pdr_id());
+  if (Pdr_index != -1) {
+    // Update the existing value
+    pdr_list_.at(Pdr_index) = *rule;
+  } else {
+    // Insert the rule
+    pdr_list_.push_back(*rule);
+  }
+  // update criteria to be updated
+  if (crit_add) session_uc->pdrs_to_install.push_back(*rule);
+}
+
+/* method to change the PDR state */
+void SessionState::set_all_pdrs(enum PdrState pdr_state) {
+  for (auto& rule : pdr_list_) {
+    rule.set_pdr_state(pdr_state);
+  }
+}
+
+int32_t SessionState::get_pdr_index(uint32_t id) {
+  int count = 0;
+  for (auto& rule : pdr_list_) {
+    if (rule.pdr_id() == id) return count;
+    count++;
+  }
+  return -1;
+}
+
+/* method to search specific pdr id existence */
+bool SessionState::search_pdr(unsigned int id) {
```

Comment:
`contains_pdr` seems better naming for this function

Reply:
Yes. changes done.